client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -306,6 +307,34 @@ private long getRemainingBytes(SegmentMetadataClientFactory metaFactory, StreamC
         return cuts;
     }
 
+    @Override
+    public CompletableFuture<Map<Stream, StreamCut>> getCurrentStreamCut(ScheduledExecutorService backgroundExecutor) {
+        String internalChkPointId = UUID.randomUUID().toString();
```

Comment:
CheckpointID are normally user defined. We should probably:
1. Restrict the character set so they cannot collide with these. (Like maybe to base64 characters, plus a few extra symbols)
2. If we scope these so that they don't collide with user defined checkpoints, we can actually infer the "Is silent" bit from the name itself and then we don't need to add / track a separate boolean.

Reply:
done.